### PR TITLE
ci: make build not failing

### LINKED ISSUE #7657: Build should fail if there are any modified files after the build (e.g. go.sum)
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The v1.5.10 build succeeded, but then created the `v1.5.10-dirty` tag due to a modified file `go.sum`, thus invalidating the release.

**Expected behavior:**
The CI should fail if there is any change to files such as go.sum.

**How to reproduce it (minimal and precise):**

See the release-1.5 build where the [git status](https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/release-1.5/259/pipeline/21) shows the following:
```
[2021-04-15T21:22:50.772Z] + git status
[2021-04-15T21:22:50.772Z] HEAD detached at b0f1a267a
[2021-04-15T21:22:50.772Z] Changes not staged for commit:
[2021-04-15T21:22:50.772Z]   (use "git add <file>..." to update what will be committed)
[2021-04-15T21:22:50.772Z]   (use "git restore <file>..." to discard changes in working directory)
[2021-04-15T21:22:50.772Z] 	modified:   go.sum
[2021-04-15T21:22:50.772Z] 
[2021-04-15T21:22:50.772Z] no changes added to commit (use "git add" and/or "git commit -a")
```

### PR DESCRIPTION
**Pull request by subhamkrai**
currently, the build action is not failing if there
is/are any modified files after the build. this commit
will validate build with `git status` after `make build`
command will fail if any changed file found.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #7657 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@travisn can you if this what you wanted?

File: go.mod [hunk 1]
```diff
@@ -30,7 +30,7 @@ require (
 	github.com/stretchr/testify v1.6.1
 	github.com/tevino/abool v1.2.0
 	github.com/yanniszark/go-nodetool v0.0.0-20191206125106-cd8f91fa16be
-	gopkg.in/ini.v1 v1.57.0
+	// gopkg.in/ini.v1 v1.57.0
```
Comment (leseb) [inline go.mod:33]: ?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -9,14 +9,21 @@ jobs:
...
     steps:
-    - name: checkout
-      uses: actions/checkout@v2
-
+    
     - uses: actions/setup-go@v2
       with:
         go-version: 1.16
 
+    - name: checkout
```
Comment (leseb) [inline .github/workflows/build.yml:17]: Why changing the position of `checkout`?

File: go.mod (hunk 1 above)
Comment (subhamkrai) [reply go.mod:33]: just for testing

File: .github/workflows/build.yml (hunk 2 above)
Comment (subhamkrai) [reply .github/workflows/build.yml:17]: this what followed in other action. example codegen, crds-gen

File: tests/scripts/validate_modified_files.sh [hunk 3]
```diff
@@ -7,6 +7,7 @@ set -ex
 CODEGEN_ERR="found codegen files! please run 'make codegen' and update your PR"
 MOD_ERR="changes found by mod.check. You may need to run make clean"
 CRD_ERR="changes found by 'make crds'. please run 'make crds' locally and update your PR"
+BUILD_ERR="changes found by make build'. please update your go.sum file"
```
Comment (travisn) [inline tests/scripts/validate_modified_files.sh:10]:
```suggestion
BUILD_ERR="changes found by make build', please commit your go.sum or other changed files"
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -9,14 +9,21 @@ jobs:
...
+      uses: actions/checkout@v2
+
+    - name: copy working directory to GOPATH
```
Comment (travisn) [inline .github/workflows/build.yml:20]: Why is this needed? The checkout doesn't put the files in the right place?

**Review by travisn — CHANGES_REQUESTED** [review]

File: tests/scripts/validate_modified_files.sh (hunk 3 above)
Comment (subhamkrai) [reply tests/scripts/validate_modified_files.sh]: Done

File: .github/workflows/build.yml (hunk 4 above)
Comment (subhamkrai) [reply .github/workflows/build.yml]: this was not needed, it's the way follow in another place (maybe putting here in same commit ya confusing). But we have removed this build action so we are good.

File: tests/scripts/validate_modified_files.sh [hunk 5]
```diff
@@ -36,6 +37,9 @@ case "$1" in
   crd)
     validate "$CRD_ERR"
   ;;
+  build)
+    validate "$BUILD_ERR"
+  ;;
   *)
     echo $"Usage: $0 {codegen|modcheck|crd}"
```
Comment (leseb) [inline tests/scripts/validate_modified_files.sh:44]:
```suggestion
    echo $"Usage: $0 {codegen|modcheck|crd|build}"
```

**Review by leseb — CHANGES_REQUESTED** [review]

File: tests/scripts/validate_modified_files.sh (hunk 5 above)
Comment (subhamkrai) [reply tests/scripts/validate_modified_files.sh]: done.

**Review by leseb — APPROVED** [review]

File: .github/workflows/canary-integration-test.yml [hunk 6]
```diff
@@ -552,6 +557,7 @@ jobs:
       run: |
         # set VERSION to a dummy value since Jenkins normally sets it for us. Do this to make Helm happy and not fail with "Error: Invalid Semantic Version"
         GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='ceph' VERSION=0 build
+        tests/scripts/validate_modified_files.sh build
```
Comment (travisn) [inline .github/workflows/canary-integration-test.yml:560]: With #7677 can this verification be added in the build helper so we won't need to call this script everywhere?

Comment (subhamkrai) [reply .github/workflows/canary-integration-test.yml:560]: @travisn yes

**Review by travisn — APPROVED** [review]
